addons/crm_mail_plugin/tests/test_crm_mail_plugin.py

```diff
@@ -54,3 +54,62 @@ def test_get_contact_data(self):
             msg="The first lead belongs to the first partner, it should be returned")
         self.assertFalse([lead for lead in result["leads"] if lead["lead_id"] == lead_2.id],
             msg="The second lead does not belong to the first partner, it should not be returned")
+
+    @mock_auth_method_outlook('employee')
+    def test_crm_lead_create_multi_company(self):
+        # create Company_A and Company_B
+        company_a = self.env['res.company'].create({
+            'name': 'Company_A',
+        })
+        company_b = self.env['res.company'].create({
+            'name': 'Company_B',
+        })
+
+        # create contact belonging to Company_B
+        contact = self.env['res.partner'].create({
+            'name': 'John Doe',
+            'email': 'john.doe@example.com',
+            'company_id': company_b.id,
+        })
+
+        # set default company to Company_A
+        self.env.user.company_id = company_a.id
+
+        # Set up the user's access to crm.lead
+        self.user_test.groups_id |= self.env.ref("sales_team.group_sale_salesman_all_leads")
+
+        # Add company_B to user_test to have access to records related to company_B
+        self.user_test.write({'company_ids': [(4, company_b.id)]})
+
+        # send request to create lead for the contact
+        data = {
+            'id': 0,
+            'jsonrpc': '2.0',
+            'method': 'call',
+            'params': {
+                'partner_id': contact.id,
+                'email_body': 'test body',
+                'email_subject': 'test subject',
+            },
+        }
+        result = self.url_open(
+            '/mail_plugin/lead/create',
+            data=json.dumps(data).encode(),
+            headers={'Content-Type': 'application/json'}
+        ).json()["result"]
```

Comment:
picky std (you mix double and single)
```suggestion
        ).json()['result']
```

(check also other place in this test)